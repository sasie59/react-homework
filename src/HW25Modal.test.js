import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW25Modal';

test('should HW25Modal', () => {
  const {container, getByTestId } = render(<HW />);
  userEvent.click(getByTestId('arr-0'));
  expect(container).toMatchSnapshot();

  userEvent.click(getByTestId('arr-1'));
  expect(container).toMatchSnapshot();

  userEvent.click(getByTestId('arr-2'));
  expect(container).toMatchSnapshot();
  
  userEvent.click(getByTestId('arr-3'));
  expect(container).toMatchSnapshot();

  userEvent.click(getByTestId('arr-4'));
  expect(container).toMatchSnapshot();
});
