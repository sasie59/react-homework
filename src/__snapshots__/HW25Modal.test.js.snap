// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW25Modal 1`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="arr-0"
      name="button"
    >
      modal
    </button>
    <button
      class="button"
      data-testid="arr-1"
      name="button"
    >
      top
    </button>
    <button
      class="button"
      data-testid="arr-2"
      name="button"
    >
      header
    </button>
    <button
      class="button"
      data-testid="arr-3"
      name="button"
    >
      animation
    </button>
    <button
      class="button"
      data-testid="arr-4"
      name="button"
    >
      top + animation
    </button>
    <div>
      <div
        class="modal-bg"
        data-testid="modal-bg"
        id="modal-bg"
      >
        <div
          class="modal-content "
        >
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ex vel in nam sint dolorum, eligendi delectus, dolor quos eum magni facere nemo fuga distinctio. Magni possimus aspernatur corrupti impedit laborum!
          <br />
          <button
            data-testid="close"
            id="button"
            name="button"
          >
            關閉
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should HW25Modal 2`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="arr-0"
      name="button"
    >
      modal
    </button>
    <button
      class="button"
      data-testid="arr-1"
      name="button"
    >
      top
    </button>
    <button
      class="button"
      data-testid="arr-2"
      name="button"
    >
      header
    </button>
    <button
      class="button"
      data-testid="arr-3"
      name="button"
    >
      animation
    </button>
    <button
      class="button"
      data-testid="arr-4"
      name="button"
    >
      top + animation
    </button>
  </div>
</div>
`;

exports[`should HW25Modal 3`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="arr-0"
      name="button"
    >
      modal
    </button>
    <button
      class="button"
      data-testid="arr-1"
      name="button"
    >
      top
    </button>
    <button
      class="button"
      data-testid="arr-2"
      name="button"
    >
      header
    </button>
    <button
      class="button"
      data-testid="arr-3"
      name="button"
    >
      animation
    </button>
    <button
      class="button"
      data-testid="arr-4"
      name="button"
    >
      top + animation
    </button>
    <div>
      <div
        class="modal-bg"
        data-testid="modal-bg"
        id="modal-bg"
      >
        <div
          class="modal-content "
        >
          <h2>
            My Header
          </h2>
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ex vel in nam sint dolorum, eligendi delectus, dolor quos eum magni facere nemo fuga distinctio. Magni possimus aspernatur corrupti impedit laborum!
          <br />
          <button
            data-testid="close"
            id="button"
            name="button"
          >
            關閉
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should HW25Modal 4`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="arr-0"
      name="button"
    >
      modal
    </button>
    <button
      class="button"
      data-testid="arr-1"
      name="button"
    >
      top
    </button>
    <button
      class="button"
      data-testid="arr-2"
      name="button"
    >
      header
    </button>
    <button
      class="button"
      data-testid="arr-3"
      name="button"
    >
      animation
    </button>
    <button
      class="button"
      data-testid="arr-4"
      name="button"
    >
      top + animation
    </button>
  </div>
</div>
`;

exports[`should HW25Modal 5`] = `
<div>
  <div>
    <button
      class="button"
      data-testid="arr-0"
      name="button"
    >
      modal
    </button>
    <button
      class="button"
      data-testid="arr-1"
      name="button"
    >
      top
    </button>
    <button
      class="button"
      data-testid="arr-2"
      name="button"
    >
      header
    </button>
    <button
      class="button"
      data-testid="arr-3"
      name="button"
    >
      animation
    </button>
    <button
      class="button"
      data-testid="arr-4"
      name="button"
    >
      top + animation
    </button>
    <div>
      <div
        class="modal-bg"
        data-testid="modal-bg"
        id="modal-bg"
      >
        <div
          class="modal-content "
        >
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ex vel in nam sint dolorum, eligendi delectus, dolor quos eum magni facere nemo fuga distinctio. Magni possimus aspernatur corrupti impedit laborum!
          <br />
          <button
            data-testid="close"
            id="button"
            name="button"
          >
            關閉
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
